sklearn/utils/validation.py

```diff
@@ -32,7 +32,28 @@
 FLOAT_DTYPES = (np.float64, np.float32, np.float16)
 
 
-def _deprecate_positional_args(f):
+def _deprecate_positional_args(version="0.25"):
+    """Decorator for methods that issues warnings for positional arguments.
+
+    Using the keyword-only argument syntax in pep 3102, arguments after the
+    * will issue a warning when passed as a positional argument.
+
+    Parameters
+    ----------
+    version : callable or str, default="0.25"
+        The version when positional arguments will result in error. If
+        callable, then "0.25" will be used for error message.
+    """
+    if callable(version):
```

Comment:
This is ugly, can't we make version keyword only to fix this, or something?

Reply:
This trick is used because of how `_deprecate_positional_args`'s first parameter can be string:

```py
@_deprecate_positional_args(version="0.26")
def f():
	pass
```

or a function:

```py
@_deprecate_positional_args
def g():
	pass
```

The "nicer" solution would be to always provide the version, but that would be a much bigger diff.

As a side note, this trick is used in Python's `lru_cache`:

https://github.com/python/cpython/blob/63298930fb531ba2bb4f23bc3b915dbf1e17e9e1/Lib/functools.py#L479